### ISSUE TITLE
Association between WallSurface and DoorSurface/WindowSurface

### ISSUE THREAD

**Issue by TatjanaKutzner** [opengeospatial/CityGML-3.0CM#54]
Currently, the UML model does not define an association between WallSurface and DoorSurface/WindowSurface. Would such an association be useful, or is it sufficient to relate these surfaces using the relation types from the Core module?

**Comment by hlg** [comment]
I believe, yes, if only for backward compatibility (`opening` association), but I did not look in detail at the relation types . See also: https://github.com/opengeospatial/CityGML-3.0CM/issues/22#issuecomment-458140152

**Comment by TatjanaKutzner** [comment]
In the web conference on 7 March 2019 we discussed that it is important to have this assocation.

We decided on making _DoorSuface_ and _WindowSurface_ subclasses of a new superclass _AbstractFillingSurface_ which in turn inherits from _AbstractSurfaceBoundary_. In addition, an association between _AbstractConstructionSurface_ and _AbstractFillingSurface_ is defined. In this way, we prevent recursions of Surfaces inside other Surfaces, and allow for DoorSurfaces and WindowSurfaces inside WallSurfaces, RoofSurfaces and all other surfaces.

Above that, we will also add an association from _AbstractConstructionSurface_ to _ClosureSurface_.

Furthermore, a class _AbstractFillingElement_ is required as superclass of _Door_ and _Window_ as well as an association from _AbstractConstructiveElement_ to _AbstractFillingElement_, to allow for relating Door and Window objects with Buildings.

We will also provide an example dataset for these modifications.
